pkg/util/validation/limits.go

```diff
@@ -305,6 +306,10 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&l.RulerRecordingRulesEvaluationEnabled, "ruler.recording-rules-evaluation-enabled", true, "Controls whether recording rules evaluation is enabled. This configuration option can be used to forcefully disable recording rules evaluation on a per-tenant basis.")
 	f.BoolVar(&l.RulerAlertingRulesEvaluationEnabled, "ruler.alerting-rules-evaluation-enabled", true, "Controls whether alerting rules evaluation is enabled. This configuration option can be used to forcefully disable alerting rules evaluation on a per-tenant basis.")
 	f.BoolVar(&l.RulerSyncRulesOnChangesEnabled, "ruler.sync-rules-on-changes-enabled", true, "True to enable a re-sync of the configured rule groups as soon as they're changed via ruler's config API. This re-sync is in addition of the periodic syncing. When enabled, it may take up to few tens of seconds before a configuration change triggers the re-sync.")
+	if l.RulerMaxRulesPerRuleGroupByNamespace.data == nil {
```

Comment:
Why wouldn't it be nil?

Also, maybe we could have a helper method like `isInitialized()`? Or maybe even store a pointer to `LimitsMap` instead of having a zero-struct and having to check an unexported field.

Reply:
> Also, maybe we could have a helper method like isInitialized()? 
Done in https://github.com/grafana/mimir/pull/8378/commits/a2e11637becfaa53cc97ba9a47f43cabe6bcb9ae

> Why wouldn't it be nil?
We wouldn't care if it's nil if you're specifying a value for the argument directly as it'll be marshalled from the JSON. When you're not, we care about having a default value so that it show up in the documentation. 

I am not sure if this is the answer to your question - if it isn't feel free to elaborate and I'll be happy to answer/fix your comment.